### ISSUE TITLE
Crash on a BINX file with ROI and fastForward = TRUE

### ISSUE THREAD

**Issue by mcol** [R-Lum/Luminescence#356]
This crashes:
```R
rectype.128.bin <- test_path("_data/bin-tests/rectype-128.binx")
res <- read_BIN2R(rectype.128.bin, verbose = FALSE, fastForward=TRUE)
# [src_create_RLumDataCurve_matrix()] BIN/BINX-file non-conform. TL curve may be wrong!
# Error in if (!is.null(object@METADATA[["RECTYPE"]]) && object@METADATA[["RECTYPE"]][x] ==  :
#   missing value where TRUE/FALSE needed
```
The bug is quite old, it was already there in 0.9.24.

**Comment by RLumSK** [comment]
@mcol ... oops. This is on me then!

**Comment by mcol** [comment]
Don't worry, I think my fix for #336 will solve that too.
